File: CafeEase/UI/cafeease_mobile/lib/models/order.dart
import 'package:json_annotation/json_annotation.dart';
import 'order_item.dart';

part 'order.g.dart';

@JsonSerializable()
class Order {
  int? id;
  String? status;
  DateTime? orderDate;
  double? totalAmount;
  int? tableId;
  int? userId;
  List<OrderItem>? items;

  Order({
    this.id,
    this.status,
    this.orderDate,
    this.totalAmount,
    this.tableId,
    this.userId,
    this.items,
  });

  factory Order.fromJson(Map<String, dynamic> json) => _$OrderFromJson(json);

  Map<String, dynamic> toJson() => _$OrderToJson(this);
}

File: CafeEase/CafeEase.Services/Database/Order.cs
using CafeEase.Model;
using System.Collections.Generic;
using System;

namespace CafeEase.Services.Database
{
    public class Order
    {
        public int Id { get; set; }

        public DateTime OrderDate { get; set; }
        public decimal TotalAmount { get; set; }
        public string Status { get; set; } = null!;
        public int UserId { get; set; }
        public virtual User User { get; set; } = null!;

        public int TableId { get; set; }
        public virtual Table Table { get; set; } = null!;

        public virtual ICollection<OrderItem> OrderItems { get; set; } = new List<OrderItem>();
    }
}
